spring-cloud-alicloud-oss/src/main/java/com/alibaba/alicloud/oss/resource/OssStorageResource.java

```diff
@@ -43,18 +46,31 @@
  * @see Bucket
  * @see OSSObject
  */
-public class OssStorageResource implements Resource {
+public class OssStorageResource implements WritableResource {
+
+	private static final Logger logger = LoggerFactory.getLogger(OssStorageResource.class);
+
+	private static final String MESSAGE_KEY_NOT_EXIST = "The specified key does not exist.";
 
 	private final OSS oss;
 	private final String bucketName;
 	private final String objectKey;
 	private final URI location;
+	private final boolean autoCreateFiles;
+
+	private static final ExecutorService executorService = new ThreadPoolExecutor(8, 128,
```

Comment:
It's hard code.

I prefer construct it in `OssAutoConfiguration` by `@ConditionalOnMissingBean`.

So developers can override it.

refer https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/task/TaskExecutionAutoConfiguration.java#L78

Reply:
And need `ConfigurableListableBeanFactory` constructor attribute in `OssStorageResource`.

We can set `ConfigurableListableBeanFactory` in `OssStorageProtocolResolver`.